File: app/src/main/java/com/robinson/notewithreminder/utilities/ReminderWorker.java
package com.robinson.notewithreminder.utilities;

import android.content.Context;

import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class ReminderWorker extends Worker {
    private Context context;

    public ReminderWorker(Context context, WorkerParameters params) {
        super(context, params);
        this.context = context;
    }

    @Override
    public Result doWork() {
        long noteId = getInputData().getLong(CommonParameters.noteId, 0);
        String noteTitle = getInputData().getString(CommonParameters.noteTitle);
        String noteDescription = getInputData().getString(CommonParameters.noteDescription);
        String noteDate = getInputData().getString(CommonParameters.noteDate);
        String noteTime = getInputData().getString(CommonParameters.noteTime);

        NotificationHelper notificationHelper = new NotificationHelper(context);
        notificationHelper.createNotification(
                noteId,
                noteTitle != null ? noteTitle : "",
                noteDescription != null ? noteDescription : "",
                noteDate != null ? noteDate : "",
                noteTime != null ? noteTime : ""
        );

        MusicControl.getInstance(context).playMusic();

        return Result.success();
    }
}

File: app/src/main/java/com/robinson/notewithreminder/utilities/ReminderWorker.kt
package com.robinson.notewithreminder.utilities

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.robinson.notewithreminder.utilities.MusicControl.Companion.getInstance

class ReminderWorker(private val context: Context, params: WorkerParameters?) : Worker(
    context, params!!
) {
    override fun doWork(): Result {
        val noteId = inputData.getLong(CommonParameters.noteId, 0)
        val noteTitle = inputData.getString(CommonParameters.noteTitle)
        val noteDescription = inputData.getString(CommonParameters.noteDescription)
        val noteDate = inputData.getString(CommonParameters.noteDate)
        val noteTime = inputData.getString(CommonParameters.noteTime)
        val notificationHelper = NotificationHelper(context)
        notificationHelper.createNotification(
            noteId,
            noteTitle ?: "",
            noteDescription ?: "",
            noteDate ?: "",
            noteTime ?: ""
        )
        getInstance(context)!!.playMusic()

        return Result.success()
    }
}